#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_inferior-octave-resync-dirs.dpatch
##                             by Rafael Laboissiere <rafael@debian.org>
##
## DP: Avoid detrimental effects on inferior-octave-mode if there is a
## DP: cd() command in ~/.octaverc

@DPATCH@

--- octave2.1-2.1.71.orig/emacs/octave-inf.el
+++ octave2.1-2.1.71/emacs/octave-inf.el
@@ -234,7 +234,10 @@

     ;; And finally, everything is back to normal.
     (set-process-filter proc 'inferior-octave-output-filter)
-    (run-hooks 'inferior-octave-startup-hook)))
+    (run-hooks 'inferior-octave-startup-hook)
+    ;; just in case, to be sure a cd in the startup file
+    ;; won't have detrimental effects
+    (inferior-octave-resync-dirs)))

 
 (defun inferior-octave-complete ()
